AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/client.py

```diff
@@ -0,0 +1,462 @@
+                serialize=_serialize_get_branch_key_version,
+                deserialize=_deserialize_get_branch_key_version,
+                config=self._config,
+                operation_name="GetBranchKeyVersion",
+            )
+        )
+
+    def get_beacon_key(self, input: GetBeaconKeyInput) -> GetBeaconKeyOutput:
+        """Get a Beacon Key from the Key Store.
+
+        :param input: Inputs for getting a Beacon Key
+        """
+        return asyncio.run(
+            self._execute_operation(
+                input=input,
+                plugins=[],
+                serialize=_serialize_get_beacon_key,
+                deserialize=_deserialize_get_beacon_key,
+                config=self._config,
+                operation_name="GetBeaconKey",
+            )
+        )
+
+    async def _execute_operation(
+        self,
+        input: Input,
+        plugins: list[Plugin],
+        serialize: Callable[[Input, Config], Awaitable[DafnyRequest]],
+        deserialize: Callable[[DafnyResponse, Config], Awaitable[Output]],
+        config: Config,
+        operation_name: str,
+    ) -> Output:
+        try:
+            return await self._handle_execution(
+                input, plugins, serialize, deserialize, config, operation_name
+            )
+        except Exception as e:
+            # Make sure every exception that we throw is an instance of ServiceError so
+            # customers can reliably catch everything we throw.
+            if not isinstance(e, ServiceError):
+                raise ServiceError(e) from e
+            raise e
+
+    async def _handle_execution(
+        self,
+        input: Input,
+        plugins: list[Plugin],
+        serialize: Callable[[Input, Config], Awaitable[DafnyRequest]],
+        deserialize: Callable[[DafnyResponse, Config], Awaitable[Output]],
+        config: Config,
+        operation_name: str,
+    ) -> Output:
+        context: InterceptorContext[Input, None, None, None] = InterceptorContext(
+            request=input,
+            response=None,
+            transport_request=None,
+            transport_response=None,
+        )
+        _client_interceptors = config.interceptors
+        client_interceptors = cast(
+            list[Interceptor[Input, Output, DafnyRequest, DafnyResponse]],
+            _client_interceptors,
+        )
+        interceptors = client_interceptors
+
+        try:
+            # Step 1a: Invoke read_before_execution on client-level interceptors
+            for interceptor in client_interceptors:
+                interceptor.read_before_execution(context)
+
+            # Step 1b: Run operation-level plugins
+            # config = deepcopy(config)
+            for plugin in plugins:
+                plugin(config)
+
+            _client_interceptors = config.interceptors
+            interceptors = cast(
+                list[Interceptor[Input, Output, DafnyRequest, DafnyResponse]],
+                _client_interceptors,
+            )
```

Comment:
Why do we need this again?

Reply:
I don't know if we need this. This is generated by Smithy-Python. I've left it in internal-only code, but sealed off all the "entry points" to these interfaces. This might be useful at some point, but I tend to doubt it. If we remove this later on, it would not be a breaking change.

Reply:
I think I worded the comment wrong. I meant that this same code block is repeated twice.

Reply:
Oh! Then the answer is:
I suspect this is useful for Smithy-Python in case a plugin modifies `config.interceptors` in the `plugin(config)` call a few lines above.